stdlib/public/Concurrency/Task.cpp

```diff
@@ -396,43 +397,70 @@ task_future_wait_resume_adapter(SWIFT_ASYNC_CONTEXT AsyncContext *_context) {
   return _context->ResumeParent(_context->Parent);
 }
 
-/// All `swift_task_create*` variants funnel into this common implementation.
+/// Implementation of task creation.
 ///
 /// If \p isAsyncLetTask is true, the \p closureContext is not heap allocated,
 /// but stack-allocated (and must not be ref-counted).
 /// Also, async-let tasks are not heap allocated, but allocated with the parent
 /// task's stack allocator.
-static AsyncTaskAndContext swift_task_create_group_future_commonImpl(
-    size_t rawFlags,
-    TaskGroup *group, // TODO: express as an option -- ktoso
+SWIFT_CC(swift)
+static AsyncTaskAndContext swift_task_create_commonImpl(
+    size_t rawTaskCreateFlags,
     TaskOptionRecord *options,
     const Metadata *futureResultType,
     FutureAsyncSignature::FunctionType *function, void *closureContext,
     size_t initialContextSize) {
-  JobFlags flags(rawFlags);
-  bool isAsyncLetTask = flags.task_isAsyncLetTask();
-  assert((futureResultType != nullptr) == flags.task_isFuture());
-  assert(!isAsyncLetTask || isAsyncLetTask == flags.task_isChildTask());
-  assert(!flags.task_isFuture() ||
-         initialContextSize >= sizeof(FutureAsyncContext));
-  assert((group != nullptr) == flags.task_isGroupChildTask());
+  TaskCreateFlags taskCreateFlags(rawTaskCreateFlags);
+
+  // Propagate task-creation flags to job flags as appropriate.
+  JobFlags jobFlags(JobKind::Task, taskCreateFlags.getPriority());
+
+  bool isAsyncLetTask = taskCreateFlags.isAsyncLetTask();
+  if (isAsyncLetTask) {
+    jobFlags.task_setIsAsyncLetTask(true);
+    jobFlags.task_setIsChildTask(true);
+  } else {
+    jobFlags.task_setIsChildTask(taskCreateFlags.isChildTask());
+  }
```

Comment:
ah yeah that's much nicer. I had initially started implementing this by asserting here but makes sense to just infer from asynclet -> must be child. 👍